rusk-wallet/src/wallet.rs

```diff
@@ -496,30 +495,11 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
             return Err(Error::NotDirectory);
         }
 
-        // get our keys for this address
-        let keys = self.provisioner_keys(profile_idx)?;
+        let (pk, sk) = self.provisioner_keys(profile_idx)?;
+        let path = PathBuf::from(dir);
+        let filename = filename.unwrap_or(profile_idx.to_string());
 
-        // set up the path
-        let mut path = PathBuf::from(dir);
-        path.push(filename.unwrap_or(profile_idx.to_string()));
-
-        // export public key to disk
-        let bytes = keys.0.to_bytes();
-        fs::write(path.with_extension("cpk"), bytes)?;
-
-        // create node-compatible json structure
-        let bls = BlsKeyPair {
-            public_key_bls: keys.0.to_bytes(),
-            secret_key_bls: keys.1.to_bytes(),
-        };
-        let json = serde_json::to_string(&bls)?;
-
-        // encrypt data
-        let mut bytes = json.as_bytes().to_vec();
-        bytes = crate::crypto::encrypt(&bytes, pwd)?;
-
-        // export key-pair to disk
-        fs::write(path.with_extension("keys"), bytes)?;
+        node_data::bls::save_consensus_keys(&path, &filename, &pk, &sk, pwd)?;
 
         Ok((path.with_extension("keys"), path.with_extension("cpk")))
```

Comment:
Assuming that the extensions are `.keys` and `.cpk` is error prone (chaning node_data will break this)
You could change node_data::bls::save_consensus_keys to return the paths and display them here